src/main/java/com/powsybl/openloadflow/NetworkCache.java

```diff
@@ -204,39 +204,29 @@ public void onUpdate(Identifiable identifiable, String attribute, String variant
                 return;
             }
             boolean done = false;
-            switch (attribute) {
-                case "v":
-                case "angle":
-                case "p":
-                case "q":
-                case "p1":
-                case "q1":
-                case "p2":
-                case "q2":
-                    // ignore because it is related to state update and won't affect LF calculation
-                    done = true;
-                    break;
-
-                default:
-                    if (identifiable.getType() == IdentifiableType.GENERATOR) {
-                        Generator generator = (Generator) identifiable;
-                        if (attribute.equals("targetV")
-                                && onGeneratorUpdate(generator, attribute, oldValue, newValue)) {
-                            done = true;
-                        }
-                    } else if (identifiable.getType() == IdentifiableType.SHUNT_COMPENSATOR) {
-                        ShuntCompensator shunt = (ShuntCompensator) identifiable;
-                        if (attribute.equals("sectionCount")
-                                && onShuntUpdate(shunt, attribute)) {
-                            done = true;
-                        }
-                    } else if (identifiable.getType() == IdentifiableType.SWITCH
-                            && attribute.equals("open")) {
-                        if (onSwitchUpdate(identifiable.getId(), (boolean) newValue)) {
-                            done = true;
-                        }
+
+            if (List.of("v", "angle", "p", "q", "p1", "q1", "p2", "q2").contains(attribute)) {
```

Comment:
why this change? this is a sequential search worst in term of performance a a switch/case

Reply:
Sonar asks for an if loop here instead of a switch case. But I understand if we revert.